ai21/clients/common/conversational_rag.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from ai21.models import ChatMessage
+from ai21.models._pydantic_compatibility import _to_dict
+from ai21.models.responses.conversational_rag_response import ConversationalRagResponse
+from ai21.models.retrieval_strategy import RetrievalStrategy
+from ai21.types import NotGiven, NOT_GIVEN
+from ai21.utils.typing import remove_not_given
+
+
+class ConversationalRag(ABC):
+    _module_name = "conversational-rag"
+
+    @abstractmethod
+    def create(
+        self,
+        messages: List[ChatMessage],
+        *,
+        path: str | NotGiven = NOT_GIVEN,
+        labels: List[str] | NotGiven = NOT_GIVEN,
+        file_ids: List[str] | NotGiven = NOT_GIVEN,
+        max_segments: int | NotGiven = NOT_GIVEN,
+        retrieval_strategy: RetrievalStrategy | NotGiven = NOT_GIVEN,
```

Comment:
In order to not depend on these exact strings the `Literal` define. I think a good approach here would be to change the type to `RetrievalStrategy | str | NotGiven`. This way if something changes in the server, we don't need to rush and change it here. wdyt?

Reply:
sounds good